### DISCUSSION TITLE
Issue with "-transparent-color" in IMv7

### DISCUSSION THREAD

**Discussion by GeeMack** [ImageMagick/ImageMagick#5297] [Development] [answered]
I'm using ImageMagick 7.1.0-40 Q16-HDRI x64 on Windows 10, and I've run across a bug when setting "-transparent-color". It looks like it will produce an error if the argument to "-transparent-color" has any alphabet characters. These commands...

    magick rose: -transparent-color black info:

    magick rose: -transparent-color #AA3333 info:

... return these errors...

    magick: invalid argument for option '-transparent-color' 'black' at CLI arg 2 @ error/operation.c/CLISettingOptionInfo/1486.

    magick: invalid argument for option '-transparent-color' '#AA3333' at CLI arg 2 @ error/operation.c/CLISettingOptionInfo/1486.

Arguments without any alphabet characters appear to work as expected. These commands...

    magick rose: -transparent-color #443333 info:

    magick rose: -transparent-color #000 info:

... return this info...

    rose: PNM 70x46 70x46+0+0 8-bit sRGB 9673B 0.000u 0:00.000

This has been an issue at least as far back as v7.1.0-35.

**Comment by snibgo** [comment]
Confirmed for v7.1.0-32 on Windows 8.1.

It is weird. "#000" works but "#abc" doesn't.

**Comment by urban-warrior** [comment, marked as answer]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/archive/beta/ by sometime tomorrow.

**Reply by GeeMack** [reply]
I have installed ImageMagick 7.1.0-43 Q16-HDRI x64, and the issue appears to be completely resolved. Thanks!
